geometry.scad

```diff
@@ -1008,19 +1011,23 @@ function closest_point_on_plane(plane, point) =
 // Returns undef if line is parallel to, but not on the given plane.
 function _general_plane_line_intersection(plane, line, eps=EPSILON) =
     let(
-        l0 = line[0],  // Ray start point
-        u = line[1] - l0,  // Ray direction vector
-        n = plane_normal(plane),
-        p0 = n * plane[3],  // A point on the plane
-        w = l0 - p0  // Vector from plane point to ray start
-    ) approx(n*u, 0, eps=eps) ? (
-        // Line is parallel to plane.
-        approx(n*w, 0, eps=eps)
-          ? [line, undef] // Line is on the plane.
-          : undef // Line never intersects the plane.
-    ) : let(
-        t = (-n * w) / (n * u)  // Distance ratio along ray
-    ) [ l0 + u*t, t ];
+        a = plane*[each line[0],-1],         //  evaluation of the plane expression at line[0] 
+        b = plane*[each(line[1]-line[0]),0]  // difference between the plane expression evaluation at line[1] and at line[0]
+    )
+    approx(b,0,eps)                          // is  (line[1]-line[0]) "parallel" to the plane ?
+    ? approx(a,0,eps)                        // is line[0] on the plane ?
+       ? [line,undef]                        // line is on the plane
+       : undef                               // line is parallel but not on the plane
+    : [ line[0]-a/b*(line[1]-line[0]), -a/b ];
+    
+    
+// Function: normalize_plane()
+// Usage: normalize_plane(plane)
```

Comment:
The function usage description (`normalize_plane(plane)`) needs to be on its own line. The text following on the same line as the `Usage:` header is an optional usage type title like "When called as a function".

```openscad
// Usage:
//   p = normalize_plane(plane);
```